src/core/evaluator.js

```diff
@@ -170,23 +170,22 @@ var PartialEvaluator = (function PartialEvaluatorClosure() {
            cs.isDefaultDecode(dict.getArray('Decode', 'D'));
   };
 
-  function PartialEvaluator(pdfManager, xref, handler, pageIndex,
-                            idFactory, fontCache, builtInCMapCache, options) {
-    this.pdfManager = pdfManager;
-    this.xref = xref;
-    this.handler = handler;
-    this.pageIndex = pageIndex;
-    this.idFactory = idFactory;
-    this.fontCache = fontCache;
-    this.builtInCMapCache = builtInCMapCache;
-    this.options = options || DefaultPartialEvaluatorOptions;
+  function PartialEvaluator(params) {
+    this.pdfManager = params.pdfManager;
```

Comment:
Excellent points, thank you for the feedback!
I've updated the PR, please check to see if this is what you had in mind.